lighter/callbacks/logger.py

```diff
@@ -0,0 +1,375 @@
+import re
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Tuple, Union, Optional
+
+import torch
+import torch.distributed as dist
+import torchvision
+from loguru import logger
+from monai.utils.module import optional_import
+from pytorch_lightning import Callback, Trainer
+from torch.utils import tensorboard
+from yaml import safe_load
+
+from lighter import LighterSystem
+
+LIGHTNING_TO_LIGHTER_STAGE = {"train": "train", "validate": "val", "test": "test"}
+OPTIONAL_IMPORTS = {}
+
+class LighterLogger(Callback):
+
+    def __init__(self, project, log_dir, log_input_as=None, log_target_as=None, log_pred_as=None, tensorboard=False, wandb=False) -> None:
+        self.project = project
+        # Only used on rank 0, the dir is created in setup().
+        self.log_dir = Path(log_dir) / project / datetime.now().strftime("%Y%m%d_%H%M%S")
+
+        self.log_as = {"input": log_input_as, "target": log_target_as, "pred": log_pred_as}
+
+        self.tensorboard = tensorboard
+        self.wandb = wandb
+
+        # Running loss. Resets at each epoch. Loss is not calculated for `test` mode.
+        self.loss = {"train": 0, "val": 0}
+        # Epoch steps. Resets at each epoch. No `test` mode step counter as loss is not calculated for it.
+        self.epoch_step_counter = {"train": 0, "val": 0}
+        # Global steps. Replaces `Trainer.global_step` because it counts optimizer steps
+        # instead of batch steps, which can be problematic when using gradient accumulation.
+        self.global_step_counter = {"train": 0, "val": 0, "test": 0}
+
+    def setup(self, trainer: Trainer, pl_module: LighterSystem, stage: str) -> None:
```

Comment:
Unfortunately not possible as PyTorch Lightning automatically passes the argument with that name. I type hinted is as a LighterSystem (which still is a LightningModule) because this logger requires some parts that are specific to LighterSystem, for example: https://github.com/project-lighter/lighter/blob/30b15fd41b5c821230dc5141020e57b3dbb68ec5/lighter/callbacks/logger.py#L239